packages/sparrowql/src/build.ts

```diff
@@ -10,321 +11,322 @@ export type Relation = {
 };
 
 type Computed = {
-    mapper: boolean;
-    required: string[];
-    result: string[];
-    perform(relative: (name: string, shouldPrefix: boolean) => string): Record<string, any>;
-  };
+  mapper: boolean;
+  required: string[];
+  result: string[];
+  perform(relative: (name: string, shouldPrefix: boolean) => string): Record<string, any>;
+};
 
-export type Options = {
+export type Options<T> = {
   aliases: Record<string, string>;
   computed: Record<string, Computed>;
   limit?: number;
-  projection?: any;
-  query?: any;
+  projection?: SchemaMember<T, ProjectionOperators | number | boolean | any>;
+  query?: FilterQuery<T>;
   relations: Relation[];
   skip?: number;
-  sort?: any;
+  sort?: SortOptionObject<T>;
   start: string;
   typeMap?: Record<string, string>;
 };
 
-type Step = {
+type Step<T> = {
   group?: any;
   limit?: number;
-  match?: any;
-  projection?: any;
+  match?: FilterQuery<T>;
+  projection?: SchemaMember<T, ProjectionOperators | number | boolean | any>;
   relation?: Relation;
   skip?: number;
-  sort?: any;
+  sort?: SortOptionObject<T>;
 }
```

Comment:
@radekmie 

Due to my lack of proficiency in Mongo (and the types themselves being not self-explanatory) I have some trouble figuring out the types here so I am not really convinced this is typed correctly. Could you please briefly look through this  so we could discuss it on tomorrow's meeting?

Also it seems our prettier/eslint is not set-up correctly? I think I have run my IDE auto-format which changed the formatting of the whole file and running `eslint` with `--fix` doesn't seem to affect the formatting at all. Will fix this before submitting the actual PR.

Reply:
First of all, I don't see any benefit from using `<T>` everywhere - [`aggregate()` is not typed anyway](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c0c8ab511249c9436289dfca3160170dd46acb24/types/mongodb/index.d.ts#L1231-L1237). Also, using MongoDB types here is not enough, as our [`query` accepts query builders as well](https://github.com/vazco/sparrowql/blob/e65bc60b6c89edcfcf3d169245cc2144486afe47/__tests__/complex/008.js#L20-L23). I'd say we'd rather start with `Record<string, 0 | 1>` for `sort`, and `Record<string, unknown>` for both `query` and `projection`.

Now, the `Step` type is a little bit different. First of all, it should be a union of objects with single non-optional properties. `group` could be made into `Record<string, unknown>` as well.